core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -143,6 +143,12 @@ class HistoryServer(
     appCache.stop()
   }
 
+  // For testing - override stop timeout used by jetty
+  private[history] def setStopTimeout(timeout: Long): Unit = {
```

Comment:
This looks to be due to the ProxyServlet handler, the way it handles requests. I am able to get around the problem by explicitly stopping the servlet handler before the test ends. No need to alter Jetty stop timeout with this change. Updated the pull request.